src/raft/raft.rs

```diff
@@ -467,13 +483,18 @@ impl<T: Storage> Raft<T> {
 
     // bcast_heartbeat sends RPC, without entries to all the peers.
     pub fn bcast_heartbeat(&mut self) {
+        let ctx = self.read_only.last_pending_request_ctx();
+        self.bcast_heartbeat_with_ctx(ctx)
+    }
+
+    pub fn bcast_heartbeat_with_ctx(&mut self, ctx: Option<Vec<u8>>) {
```

Comment:
Two cases:
1. context field is not set, and `has_context` returns false
2. context field is set to `""` somehow(maybe a bug or data corruption on network), but it doesn't work as a read index request context. 
in L955, the usage of `get_context().is_empty()` covers both cases.

The usage of Option in L490, is to make sure only case 1 happens in heartbeat broadcast. It's strict, and good for understanding, since there is no chance a empty context "" floating around.